source/common/http/conn_manager_impl.h

```diff
@@ -521,6 +577,12 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
     absl::optional<Router::RouteConstSharedPtr> cached_route_;
     absl::optional<Upstream::ClusterInfoConstSharedPtr> cached_cluster_info_;
     std::list<DownstreamWatermarkCallbacks*> watermark_callbacks_{};
+    // Stores metadata added in the decoding filter that is being processed. Will be cleared before
```

Comment:
I notice that we have similar storage on a per-filter basis as well as on a connection basis. Is it possible to collapse that somehow? If not can you add some clarifying comments on why we need both types of storage? It's not immediately clear.

Reply:
Added comments and removed not used variables.